src/generators/android/android.ts

```diff
@@ -0,0 +1,328 @@
+
+	const serializeArray =
+		`${Separator.Indent}List<?> p = TypewriterUtils.serialize(${name});\n` +
+		`${Separator.NewLineIndent}${defaultHandler(rawName, 'p', false)}`
+
+	return isArrayType && isArrayType[1] !== 'Properties'
+		? serializeArray
+		: isSerializable
+		? serializeObject
+		: defaultHandler()
+}
+
+const getValidParams = (potentialParams: Param[]) =>
+	potentialParams.reduce((acc: string[], { hasParam, name, type }) => {
+		if (hasParam) {
+			acc.push(`${Modifier.FinalNullable} ${type} ${name}`)
+		}
+		return acc
+	}, [])
+
+const getValidArgs = (potentialArgs: Arg[], defaultArg?: string[]) =>
+	potentialArgs.reduce((acc: string[], { inUse, execution, fallback }) => {
+		if (inUse) {
+			acc.push(execution)
+		} else if (fallback) {
+			acc.push(fallback)
+		}
+		return acc
+	}, defaultArg || [])
+
+const intersperse = (values: string[], char: string) => {
+	let joined = values[0] || ''
+	if (values.length) {
+		for (let i = 1; i < values.length; i++) {
+			joined += `${char}${values[i]}`
+		}
+	}
+	return joined
+}
+
+function generateFunctionSignature(
+	{ functionName, propsParam }: { functionName: string; propsParam: boolean },
+	withOptions: boolean
+): string {
+	const params = getValidParams([
+		{ hasParam: propsParam, name: 'props', type: upperFirst(functionName) },
+		{ hasParam: withOptions, name: 'options', type: 'Options' },
+	])
+
+	return `public void ${functionName}(${intersperse(params, Separator.Comma)})`
+}
+
+function generateFunctionExecution(
+	{ rawEventName, propsParam }: { rawEventName: string; propsParam: boolean },
+	withOptions: boolean
+): string {
+	const args = getValidArgs(
+		[
+			{ inUse: propsParam, execution: Properties.ToProperties, fallback: Properties.Create },
+			{ inUse: withOptions, execution: Options.MergeOptions, fallback: Options.Create },
+		],
+		[`"${rawEventName}"`]
+	)
+	return `{
+    this.analytics.track(${intersperse(args, Separator.Comma)});
+  }`
+}
+
+function generatePropertiesGetterSetter(name: string): string {
+	return `
+  private ${name}(Properties properties) {
+    this.properties = properties;
+	}
+
+  protected Properties toProperties() {
+    return properties;
+	}
+  `
+}
+
+function shouldThrowRuntimeError({
+	isRequired,
+	nonNullPayload,
+}: AndroidPropertyContext & BasePropertyContext) {
+	return isRequired && nonNullPayload
+}
+
+function generateRequiredPropertyRuntimeError(
+	className: string,
+	property: { rawName: string; isPayloadFieldNullable: boolean }
+): string | undefined {
+	return `if(properties.get("${property.rawName}") == null){
+    ${Separator.Indent}${
+		Separator.Indent
+	}throw new IllegalArgumentException("${className} missing required property: ${
+		property.rawName
+	}");
+    ${Separator.Indent}}
+  `
+}
+
+function generateComment({ rawEventName }: { rawEventName: string }) {
+	const eventClass = rawEventName.replace(/\s|\_/g, '')
+	return `* @param props {@link ${eventClass}} to add extra information to this call.
```

Comment:
Can this live in the template too? And instead of using the raw event name, you might want to use the `name` field itself since the namer will also remove invalid fields, etc.